reference/src/glossary.md

```diff
@@ -0,0 +1,55 @@
+## Glossary
+
+#### Interior mutability
+
+*Interior Mutability* means mutating memory where there also exists a live shared reference pointing to the same memory.
+This propagates recursively through references, but not through raw pointers.
+So, for example, if data pointed to by a `&T` or `& &mut T` is mutated, that's interior mutability.
```

Comment:
"Data pointed to" is a bit ambiguous when talking about pointers-to-pointers, maybe talk more specifically about the data contained in the `T`?

Reply:
Is "immediately pointed to" better? I think we should have some terminology for this that's purely operational and does not refer to types.

Reply:
Done.